builder/dockerfile/parser/parser.go

```diff
@@ -35,12 +37,29 @@ type Node struct {
 }
 
 var (
-	dispatch              map[string]func(string) (*Node, map[string]bool, error)
-	tokenWhitespace       = regexp.MustCompile(`[\t\v\f\r ]+`)
-	tokenLineContinuation = regexp.MustCompile(`\\[ \t]*$`)
-	tokenComment          = regexp.MustCompile(`^#.*$`)
+	dispatch                    map[string]func(string) (*Node, map[string]bool, error)
+	tokenWhitespace             = regexp.MustCompile(`[\t\v\f\r ]+`)
+	tokenLineContinuation       *regexp.Regexp
+	tokenEscape                 rune
+	tokenEscapeCommand          = regexp.MustCompile(`^escape.*$`)
+	tokenComment                = regexp.MustCompile(`^#.*$`)
+	builderInstructionProcessed bool
+	directiveEscapeSeen         bool
 )
 
+const defaultTokenEscape = "\\"
+
+// setTokenEscape sets the default token for escaping characters in a Dockerfile.
+func setTokenEscape(s string) error {
+	if s != "`" && s != "\\" {
+		return fmt.Errorf("Invalid ESCAPE '%s'. Must be ` or \\.", s)
```

Comment:
By convention, Go errors are not supposed to be capitalized or end with punctuation: https://github.com/golang/go/wiki/CodeReviewComments#error-strings